File: Web_Client/src/app/services/Company.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Router } from '@angular/router';
import { Company } from '../models/Company';


@Injectable({
    providedIn: 'root'
})

export class CompanyService {
    private baseUrl = "https://localhost:7247/api";
    constructor(private http: HttpClient) { }

    getTopCompanies(): Observable<Company[]> {
        return this.http.get<Company[]>(`${this.baseUrl}/Company/get-top-companies`)
    }
    getAllCompanies(): Observable<Company[]> {
        return this.http.get<Company[]>(`${this.baseUrl}/Company/get-all-companies`)
    }
    getCompaniesByName(name : string): Observable<Company[]>{
        return this.http.get<Company[]>(`${this.baseUrl}/Company/get-companies-by-name/${name}`)
    }
}

File: Web_Sever/Web_Server/Services/CompanyService.cs
using Web_Server.Interfaces;
using Web_Server.Models;

namespace Web_Server.Services
{
    public class CompanyService : ICompanyService
    {
        private readonly ICompanyRepository _companyRepository;

        public CompanyService(ICompanyRepository companyRepository)
        {
            _companyRepository = companyRepository;
        }
        public async Task<List<Company>> GetAllCompanies()
        {
            return await _companyRepository.GetAllCompanies();
        }

        public async Task<List<Company>> GetTop4Companies()
        {
            return await _companyRepository.GetTop4Companies();
        }
        public async Task<List<Company>> GetCompaniesByName(string name)
        {
            return await _companyRepository.GetCompaniesByName(name);
        }
    }
}
